include/app-framework/Buffers/DequeBuffer.hh

```diff
@@ -0,0 +1,45 @@
+#ifndef app_framework_Buffers_DequeBuffer_hh
+#define app_framework_Buffers_DequeBuffer_hh
+
+#include "app-framework-base/Buffers/Buffer.hh"
+
+#include <unistd.h>
+#include <deque>
+#include <functional>
+#include <mutex>
+
+namespace appframework{
+
+  template <class T>
+  class DequeBuffer : public BufferInput<T>, BufferOutput<T> {
+    
+  public:
+
+    DequeBuffer();
+
+    void Configure();
+    
+    size_t size() { return fDeque.size(); }
```

Comment:
This is dangerous because fDeque could be locked, so every access of fDeque should be locked. I suggest we have size counter which is atomic if order to deal with this.